File: backend/src/main/java/com/example/backend/models/Film.java
package com.example.backend.models;

import com.example.backend.enums.Genere;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "films")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Film {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String titolo;

    @Column(nullable = false)
    private String regista;

    @Column(nullable = false)
    private int annoUscita;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Genere genere;

    @Column(nullable = false)
    private String descrizione;

    @Column(nullable = true)
    private Double valutazione;
}

File: frontend/src/app/models/Film.ts
import { Genere } from "./enums/Genere";

export class Film {
    id?: number;
    titolo: string | null;
    regista: string | null;
    annoUscita: number | null;
    genere: Genere | null;
    descrizione: string | null;
    valutazione?: number | null; 
  
    constructor(
      id: number,
      titolo: string,
      regista: string,
      annoUscita: number,
      genere: Genere,
      descrizione: string,
      valutazione?: number
    ) {
      this.id = id;
      this.titolo = titolo;
      this.regista = regista;
      this.annoUscita = annoUscita;
      this.genere = genere;
      this.descrizione = descrizione;
      this.valutazione = valutazione;
    }
  }
  